plugins/woocommerce/src/Database/Migrations/CustomOrderTable/CLIRunner.php

```diff
@@ -404,53 +405,64 @@ public function verify_cot_data( $args = array(), $assoc_args = array() ) {
 			$failed_ids_in_current_batch = $this->verify_meta_data( $order_ids, $failed_ids_in_current_batch );
 			$failed_ids                  = $verbose ? array() : $failed_ids + $failed_ids_in_current_batch;
+			$error_processing            = $error_processing || ! empty( $failed_ids_in_current_batch );
 			$processed                  += count( $order_ids );
 			$batch_total_time            = microtime( true ) - $batch_start_time;
 			$batch_count ++;
 			$total_time += $batch_total_time;
 
-			if ( $verbose && count( $failed_ids_in_current_batch ) > 0 ) {
-				$errors = wp_json_encode( $failed_ids_in_current_batch, JSON_PRETTY_PRINT );
-				WP_CLI::warning(
-					sprintf(
-					/* Translators: %1$d is number of errors and %2$s is the formatted array of order IDs. */
-						_n(
-							'%1$d error found: %2$s. Please review the error above.',
-							'%1$d errors found: %2$s. Please review the errors above.',
-							count( $failed_ids_in_current_batch ),
-							'woocommerce'
-						),
-						count( $failed_ids_in_current_batch ),
-						$errors
-					)
-				);
-				if ( $remigrate ) {
+			if ( count( $failed_ids_in_current_batch ) > 0 ) {
+				if ( $verbose ) {
+					$errors = wp_json_encode( $failed_ids_in_current_batch, JSON_PRETTY_PRINT );
 					WP_CLI::warning(
 						sprintf(
-							__( 'Attempting to remigrate...', 'woocommerce' )
+						/* Translators: %1$d is number of errors and %2$s is the formatted array of order IDs. */
+							_n(
+								'%1$d error found: %2$s. Please review the error above.',
+								'%1$d errors found: %2$s. Please review the errors above.',
+								count( $failed_ids_in_current_batch ),
+								'woocommerce'
+							),
+							count( $failed_ids_in_current_batch ),
+							$errors
 						)
 					);
-					$failed_ids = array_keys( $failed_ids_in_current_batch );
-					$this->synchronizer->process_batch( $failed_ids );
-					$errors_in_remigrate_batch = $this->post_to_cot_migrator->verify_migrated_orders( $failed_ids );
-					$errors_in_remigrate_batch = $this->verify_meta_data( $failed_ids, $errors_in_remigrate_batch );
+				}
+
+				if ( $remigrate ) {
+					$failed_ids       = $failed_ids ? array_diff_key( $failed_ids, $failed_ids_in_current_batch ) : array();
+					$error_processing = ( ! $verbose ) && $failed_ids;
+
+					$verbose && WP_CLI::warning( sprintf( __( 'Attempting to remigrate...', 'woocommerce' ) ) );
+
+					$failed_ids_in_current_batch_keys = array_keys( $failed_ids_in_current_batch );
+					$this->synchronizer->process_batch( $failed_ids_in_current_batch_keys );
+					$errors_in_remigrate_batch = $this->post_to_cot_migrator->verify_migrated_orders( $failed_ids_in_current_batch_keys );
+					$errors_in_remigrate_batch = $this->verify_meta_data( $failed_ids_in_current_batch_keys, $errors_in_remigrate_batch );
+
 					if ( count( $errors_in_remigrate_batch ) > 0 ) {
+						$error_processing = true;
```

Comment:
In this case, when re-migration fails, perhaps we should still show a message that re-migration was attempted but was failed. wdyt? currently, based on messages it looks like re-migration was never attempted for failed cases.

Reply:
I've added a `remigrate_failed => true` line to the errors when that happens (see 19dc888782024cd88ff97c1946faa3212f7ff9d0). I believe this conveys the same without us having to split errors between failed and failed after remigration, but please let me know what you think.

Example:

<img width="759" alt="Screenshot 2024-02-29 at 16 14 50" src="https://github.com/woocommerce/woocommerce/assets/184724/2c9e09fb-bb51-4ffe-86e4-f1f567c02e2c">